volatility3/framework/plugins/windows/idt.py

```diff
@@ -0,0 +1,295 @@
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.PluginRequirement(
+                name="modules", plugin=modules.Modules, version=(1, 0, 0)
+            ),
+        ]
+
+    def get_module(
+        self,
+        context: interfaces.context.ContextInterface,
+        layer_name: str,
+        symbol_table: str,
+        offset: int,
+    ):
+        try:
+            mods = modules.Modules.list_modules(context, layer_name, symbol_table)
+
+            for mod in mods:
+                if mod.DllBase + mod.SizeOfImage >= offset and mod.DllBase <= offset:
+                    return mod
+        except:
+            pass
+
+        return None
+
+    @staticmethod
+    def get_section_name(ntkrnlmp, layer_name, mod, addr):
+        """Get the name of the PE section containing
+        the specified address.
+
+        @param ntkrnlmp: ntkrnlmp module object
+        @param layer_name: kernel layer name
+        @param mod: an _LDR_DATA_TABLE_ENTRY
+        @param addr: virtual address to lookup
+
+        @returns string PE section name
+        """
+
+        def name_array_to_str(name_array):
+            name = ""
+            for char in name_array:
+                if char <= 0:
+                    break
+                name += chr(char)
+            return name
+
+        try:
+            dos_header = ntkrnlmp.object(
+                object_type="_IMAGE_DOS_HEADER",
+                layer_name=layer_name,
+                offset=mod.DllBase,
+                absolute=True,
+            )
+            nt_header = dos_header.get_nt_header()
+        except ValueError:
+            return ''
+
+        for sec in nt_header.get_sections():
+            if (
+                addr > mod.DllBase + sec.VirtualAddress
+                and addr < sec.Misc.VirtualSize + (mod.DllBase + sec.VirtualAddress)
+            ):
+                return name_array_to_str(sec.Name) or ""
+
+        return ''
+
+    def get_pcrs(self, ntkrnlmp, layer_name, symbol_table):
+        # Get the number of processors
+        cpu_count_offset = ntkrnlmp.get_symbol("KeNumberProcessors").address
+        cpu_count = ntkrnlmp.object(
+            object_type="unsigned int", layer_name=layer_name, offset=cpu_count_offset
+        )
+
+        for cpu_index in range(cpu_count):
+            # Calculate the address of KiProcessorBlock
+            KiProcessorBlock_addr = (
+                ntkrnlmp.get_symbol("KiProcessorBlock").address + cpu_index * 4
+            )
+            KiProcessorBlock = ntkrnlmp.object(
+                object_type="pointer",
+                layer_name=layer_name,
+                offset=KiProcessorBlock_addr,
+            )
+
+            # Get kpcr object
+            kpcr_offset = ntkrnlmp.get_type("_KPCR").relative_child_offset("PrcbData")
```

Comment:
#1280 suggest that `PrcbData` may not always be available in all versions, this should probably either test for it, or be wrapped in a protective try/except to avoid errors making their way back to the user.